### PR TITLE
Add appeal model and event

### PR DESCRIPTION
**Pull request by krauselukas**
Extracted from https://github.com/openSUSE/open-build-service/pull/15319

This PR implement the Appeal Model and the corresponding Event required to trigger the notifications.

### CODE REVIEW

**Comment by krauselukas** [comment]
[Allow users to appeal the moderators' decision (L)](https://trello.com/c/Do5xvurq/2374-allow-users-to-appeal-the-moderators-decision-l)

File: src/api/app/models/event/appeal_created.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Event::AppealCreated < Base
```
[the remark this answers is not in the document]
Comment (krauselukas) [reply src/api/app/models/event/appeal_created.rb]: @dmarcoux done

**Comment by dmarcoux** [comment]
I have added the previously missing foreign key constraints which were flagged by `database_consistency`

>ForeignKeyChecker fail Appeal appellant should have foreign key in the database
>ForeignKeyChecker fail Appeal decision should have foreign key in the database

**Comment by dmarcoux** [comment]
The check `Isolated Migrations / check_changed_files` is wrong, the migration can be shipped with the code changes in this case.

**Comment by eduardoj** [comment]
> The check `Isolated Migrations / check_changed_files` is wrong, the migration can be shipped with the code changes in this case.

I see. The code changes doesn't make the application to be able to use the introduced migration changes.

**Review by eduardoj — APPROVED** [review]

**Pull request merged**